src/DiamondRootOval.sol

```diff
@@ -19,23 +18,48 @@ abstract contract DiamondRootOval is IBaseController, IOval, IBaseOracleAdapter
      */
     function getLatestSourceData() public view virtual returns (int256, uint256);
 
+    /**
+     * @notice Returns the requested round data from the source.
+     * @dev If the source does not support rounds this would return uninitialized data.
+     * @param roundId The roundId to retrieve the round data for.
+     * @return answer Round answer in 18 decimals.
+     * @return updatedAt The timestamp of the answer.
+     */
+    function getSourceDataAtRound(uint256 roundId) public view virtual returns (int256, uint256);
```

Comment:
question about type: chainlink uses uint80 for roundId (you use this at other parts of the code). do we know about other oracle types that would use things bigger than uint80 for their round ID? Pyth? it is likely better/more future proof to stick with the lager integer but curious I've you've seen others that need bigger.

Reply:
have not seen other round implementations than Chainlink, just using uint256 internally to be future proof for any source adapters that might have different types

Reply:
ye, and functionally it'll use the same storage & cost (full slot)